internal/gitlab/gitlab.go

```diff
@@ -1900,3 +1906,18 @@ func CreateMergeRequestCommitDiscussion(project string, mrID int, sha string, ne
 	note := discussion.Notes[0]
 	return fmt.Sprintf("%s/merge_requests/%d#note_%d", p.WebURL, note.NoteableIID, note.ID), nil
 }
+
+// hasNextPage is a pagination helper to handle the case where the API
+// response comes without the HTTP X-Total(-Page) header. Reference:
+// https://docs.gitlab.com/ee/user/gitlab_com/index.html#pagination-response-headers
+func hasNextPage(resp *gitlab.Response) (int, bool) {
+	if resp.TotalPages != 0 {
+		if resp.CurrentPage >= resp.TotalPages {
+			return 0, false
+		}
+	} else if resp.CurrentPage >= resp.NextPage {
```

Comment:
What happens in the case of, let's say, I have an API with 10 pages, the ninth iteration will return return `resp.NextPage` as 9, and on the tenth? I'm assuming NextPage will be zero by then so we return 0 and end up not retrieving page 10?

Reply:
I'm not sure if I follow your concern. The X-Page starts at `1`, so in the 9th iteration `X-Page: 9` and `X-Next-Page: 10`, so we request the `X-Page: 10`, which is the last one. With that, `resp.CurrentPage == resp.NextPage`. In theory, `CurrentPage` will never be `>` than `NextPage`.

Reply:
Ah okay.  I think both @zampierilucas were confused by the numbering.  That make sense now.

Reply:
Yeah, Thanks @bmeneguele